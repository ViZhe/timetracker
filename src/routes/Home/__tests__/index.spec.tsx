
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import { Home } from '../index';
import styles from './index.module.css';


describe('<Home />', () => {
  const props = {
    user: {
      name: null,
    },
  };

  describe('<Home /> initial', () => {
    const home: ShallowWrapper = shallow(<Home {...props} />);

    it('should match snapshot', () => {
      expect(home).toMatchSnapshot();
    });

    it('should match hello text', () => {
      expect(home.find(`.${styles.header}`).text()).toEqual('Hello, anonymous!');
    });
  });

  describe('<Home /> with props user.name', () => {
    const nextProps = {
      ...props,
      user: {
        name: 'Test name',
      },
    };

    const home:ShallowWrapper = shallow(<Home {...nextProps} />);

    it('should match snapshot', () => {
      expect(home).toMatchSnapshot();
    });

    it('should match hello text', () => {
      expect(home.find(`.${styles.header}`).text()).toEqual(`Hello, ${nextProps.user.name}!`);
    });
  });
});
