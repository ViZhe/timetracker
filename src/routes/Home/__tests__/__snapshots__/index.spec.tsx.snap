// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> <Home /> initial should match snapshot 1`] = `
<div
  className="box"
>
  <div
    className="header"
  >
    Hello, 
    anonymous
    !
  </div>
  <div
    className="body"
  >
    Let's start tracking you time together ;)
  </div>
  <div
    className="footer"
  >
    <Link
      replace={false}
      to="/time"
    >
      <Button
        block={false}
        ghost={false}
        icon="play-circle"
        loading={false}
        size="large"
        type="primary"
      >
        Start
      </Button>
    </Link>
  </div>
</div>
`;

exports[`<Home /> <Home /> with props user.name should match snapshot 1`] = `
<div
  className="box"
>
  <div
    className="header"
  >
    Hello, 
    Test name
    !
  </div>
  <div
    className="body"
  >
    Let's start tracking you time together ;)
  </div>
  <div
    className="footer"
  >
    <Link
      replace={false}
      to="/time"
    >
      <Button
        block={false}
        ghost={false}
        icon="play-circle"
        loading={false}
        size="large"
        type="primary"
      >
        Start
      </Button>
    </Link>
  </div>
</div>
`;
